arm-servicefabric/2016-09-01/swagger/servicefabric.json

```diff
@@ -0,0 +1,930 @@
+        },
+        "upgradeDomainTimeout": {
+          "type": "string"
+        },
+        "healthPolicy": {
+          "$ref": "#/definitions/PaasClusterUpgradeHealthPolicy"
+        },
+        "deltaHealthPolicy": {
+          "$ref": "#/definitions/PaasClusterUpgradeDeltaHealthPolicy"
+        }
+      }
+    },
+    "DiagnosticsStorageAccountConfig": {
+      "description": "Diagnostics storage account config",
+      "properties": {
+        "storageAccountName": {
+          "type": "string",
+          "readOnly": true
+        },
+        "primaryAccessKey": {
+          "type": "string",
+          "readOnly": true
+        },
+        "secondaryAccessKey": {
+          "type": "string",
+          "readOnly": true
+        },
+        "protectedAccountKeyName": {
+          "type": "string",
+          "readOnly": true
+        },
+        "blobEndpoint": {
+          "type": "string",
+          "readOnly": true
+        },
+        "queueEndpoint": {
+          "type": "string",
+          "readOnly": true
+        },
+        "tableEndpoint": {
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ClusterProperties": {
+      "description": "The cluster resource properties",
+      "properties": {
+        "availableClusterVersions": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ClusterVersionDetails"
+          }
+        },
+        "clusterId": {
+          "readOnly": true,
+          "type": "string"
+        },
+        "clusterState": {
+          "readOnly": true,
+          "type": "string",
+          "enum": [
+            "Default",
+            "WaitingForNodes",
+            "Deploying",
+            "BaselineUpgrade",
+            "UpdatingUserConfiguration",
+            "UpdatingUserCertificate",
+            "UpdatingInfrastructure",
+            "EnforcingClusterVersion",
+            "UpgradeServiceUnreachable",
+            "Deleting",
+            "ScaleUp",
+            "ScaleDown",
+            "AutoScale",
+            "Ready",
+            "Failed"
+          ]
+        },
+        "clusterEndpoint": {
+          "type": "string",
+          "readOnly": true
+        },
+        "clusterCodeVersion": {
+          "type": "string"
+        },
+        "certificate": {
+          "$ref": "#/definitions/CertificateDescription"
+        },
+        "reliabilityLevel": {
+          "type": "string",
+          "enum": [
+            "Invalid",
+            "Bronze",
+            "Silver",
+            "Gold",
+            "Platinum"
+          ]
+        },
+        "upgradeMode": {
+          "type": "string",
+          "enum": [
+            "Default",
+            "Automatic",
+            "Manual"
+          ]
+        },
+        "clientCertificateThumbprints": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClientCertificateThumbprint"
+          }
```

Comment:
Suppose user wants to add a `ClientCertificateThumbprint` instance to this collection via PATCH, which one of the following flow is correct?

Flow1

0. clusterPropertiesUpdateParameters = new ClusterPropertiesUpdateParameters();
1. User gets the `cluster` instance 
2. set clusterPropertiesUpdateParameters.clientCertificateThumbprints = cluster.clientCertificateThumbprints;
3. clusterPropertiesUpdateParameters.clientCertificateThumbprints.add(newClientCertificateThumbprint);
4. PATCH clusterPropertiesUpdateParameters


Flow2:

0. clusterPropertiesUpdateParameters = new ClusterPropertiesUpdateParameters();
1. clusterPropertiesUpdateParameters.clientCertificateThumbprints.add(newClientCertificateThumbprint);
2. PATCH clusterPropertiesUpdateParameters

Asking to confirm, incase of PATCH - whether sending an array with new items appends that to existing array in the server or it overwrites the previous collection.

It will be helpful to indicate this in the description, i.e. in general how the array needs to be PATCH-ed